### PR TITLE
Api typings

### PR DESCRIPTION
**Pull request by ruggi99**
## What kind of change does this PR introduce?

Upgrade.
Add return types to some APIs used for self-host.
This should help remain in sync with platform APIs.
When platform APIs are updated, not always the changes are reflected into self-host APIs.
So when changes are made on types, also placeholders for self-host should be updated accordingly.
Some type changes are assumptions based on returned data from platform.

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by ruggi99** [comment]
@sweatybridge also this when you have time and if it is something interesting to the team

File: studio/pages/api/props/project/[ref]/settings.ts [hunk 1]
```diff
@@ -31,20 +33,46 @@ const handleGetAll = async (req: NextApiRequest, res: NextApiResponse) => {
       db_ssl: false,
       db_user: 'postgres',
       jwt_secret: '-',
+      services: [
```
Comment (sweatybridge) [inline studio/pages/api/props/project/[ref]/settings.ts:36]: Need to double check on platform if this is necessary.

File: studio/data/config/project-settings-query.ts [hunk 2]
```diff
@@ -35,7 +35,6 @@ export type Service = {
 }
 
 export type AppConfig = {
-  protocol: 'https' | 'http'
```
Comment (sweatybridge) [inline studio/data/config/project-settings-query.ts:38]: Not sure if it's safe to remove

File: studio/data/config/project-api-query.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ export type AutoApiService = Service & {
   protocol: 'https' | 'http'
   endpoint: string
   restUrl: string
-  project: Project
+  project: Pick<Project, "ref">
```
Comment (sweatybridge) [inline studio/data/config/project-api-query.ts:54]: A bit strange if only ref is returned for this project. Need to double check on platform

**Review by sweatybridge — COMMENTED** [review]
Mostly ok but I don't want to overwhelm frontend these few weeks unless it's an urgent fix. I will keep this PR in mind for future reference

**Pull request closed without merging**